experimental/GModule/GModule.jl

```diff
@@ -20,7 +21,18 @@ function GAP.gap_to_julia(::Type{QabElem}, a::GAP.GapObj) #which should be a Cyc
   return z
 end
 
+function GAP.gap_to_julia(::Type{QabElem}, a::Integer)
+  E = abelian_closure(QQ)[1]
+  return E(fmpz(a))
+end
+
 (::QabField)(a::GAP.GapObj) = GAP.gap_to_julia(QabElem, a)
+Hecke.data(a::QabElem) = a.data
+
+function Hecke.number_field(::FlintRationalField, chi::Oscar.GAPGroupClassFunction; cached::Bool = false)
+  return number_field(QQ, map(x->GAP.gap_to_julia(QabElem, x), chi.values), cached = cached)
```

Comment:
With the above `QabElem` constructor, you can rewrite this as
```suggestion
  return number_field(QQ, map(QabElem, chi.values), cached = cached)
```